hbase-hbtop/pom.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <!--
+  /**
+   * Licensed to the Apache Software Foundation (ASF) under one
+   * or more contributor license agreements.  See the NOTICE file
+   * distributed with this work for additional information
+   * regarding copyright ownership.  The ASF licenses this file
+   * to you under the Apache License, Version 2.0 (the
+   * "License"); you may not use this file except in compliance
+   * with the License.  You may obtain a copy of the License at
+   *
+   *     http://www.apache.org/licenses/LICENSE-2.0
+   *
+   * Unless required by applicable law or agreed to in writing, software
+   * distributed under the License is distributed on an "AS IS" BASIS,
+   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   * See the License for the specific language governing permissions and
+   * limitations under the License.
+   */
+  -->
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>hbase</artifactId>
+    <groupId>org.apache.hbase</groupId>
+    <version>3.0.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>hbase-hbtop</artifactId>
+  <name>Apache HBase - HBTop</name>
+  <description>A real-time monitoring tool for HBase like Unix top command</description>
```

Comment:
s/like Unix/like the Unix/ or like Unix's top...

Nice that you put it in its own module.

One thought is that maybe this belongs in the hbase-operator-tools repo?

Reply:
> s/like Unix/like the Unix/ or like Unix's top...

Thank you, will modify that.

> One thought is that maybe this belongs in the hbase-operator-tools repo?

I thought so actually. However, considering to backport this to branch-1 after committing it to branch-2+, I think it's better to put this into the hbase project because managing the compatibilities is much easier. Also, I'm thinking about adding more metrics to ClusterMetrics to add more metrics to hbtop as hbtop gets the metrics from ClusterMetrics. And this will need server side modifications. In this case, I think it's better to have hbtop in the hbase project too because of the compatibility reason. What do you think? @saintstack

Reply:
bq. ....because managing the compatibilities is much easier. 

Smile. This might not be a good argument for keeping hbtop in hbase.

Being external in hbase-operator-tools, the change turnaround will be better, hbtop being able to run externally against public APIs will make for a more robust product I think -- robust against internal changes. hbase codebase is too big as is. Would be good to not add more if possible. You've done the hard work of putting it into its own module so export might be pretty smooth?

But if you think it a PITA being out in hbase-operator-tools we could commit to hbase now and then move it out if you think that'll help speed dev. This is a nice feature. Would be good to get it in.

Reply:
I suppose if it is going to evolve then yeah, you might want to keep it in hbase so hbtop ships w/ the hbase it works against.